src/otaclient/boot_control/_grub.py

```diff
@@ -871,6 +876,7 @@ def on_operation_failure(self):
     def pre_update(self, version: str, *, standby_as_ref: bool, erase_standby=False):
         try:
             logger.info("grub_boot: pre-update setup...")
+            self._update_version = version
```

Comment:
It might be better to pass it as an argument like the following, rather than storing it in a class variable:
```
def pre_update(self, *, standby_as_ref: bool, erase_standby=False):
def post_update(self, version: str):
```

Reply:
! Good catch ! I should really not do it like this, let me fix it up :innocent:

Reply:
Fixed in ca4863e26947287943a55d211b1535368118de85.